_posts/en/newsletters/2020-10-21-newsletter.md

```diff
@@ -0,0 +1,183 @@
+release candidates.*
+
+FIXME:harding to update Tuesday
+
+## Notable code and documentation changes
+
+*Notable changes this week in [Bitcoin Core][bitcoin core repo],
+[C-Lightning][c-lightning repo], [Eclair][eclair repo], [LND][lnd repo],
+[Rust-Lightning][rust-lightning repo], [libsecp256k1][libsecp256k1 repo],
+[Hardware Wallet Interface (HWI)][hwi repo], [Bitcoin Improvement Proposals
+(BIPs)][bips repo], and [Lightning BOLTs][bolts repo].*
+
+- [Bitcoin Core #19953][] implements [schnorr signature
+  verification][topic schnorr signatures] ([BIP340][]), [taproot][topic
+  taproot] verification ([BIP341][]), and [tapscript][topic tapscript]
+  verification ([BIP342][]).  The new soft fork rules are currently only
+  used in the private regression test mode (regtest); they are not
+  enforced in mainnet, testnet, or the default signet.  The plan is to
+  release the code in Bitcoin Core 0.21.0 and then prepare a subsequent
+  release (e.g. 0.21.1) that can begin enforcing the soft fork's new
+  rules when an activation signal is detected.  The particular
+  activation signal or signals to use are still being discussed in the
+  [##taproot-activation][] IRC chatroom ([logs][##taproot-activation
+  logs]).
+
+    {% comment %}<!-- $ git diff 450d2b2371...865d2c37e2 | grep '^+' \
+                        | grep -v '^+$' | grep -v '^+ */[/*]' | grep -v '^+ *\*' \
+                        | grep -v '^+++' | wc -l
+                      512
+    -->{% endcomment %}
+
+    The code [consists][sipa summary] of about [700 lines][sipa
+    consensus] of consensus-related changes (500 excluding comments and
+    whitespace) and [2,100 lines][sipa tests] of tests.  Over 30 people
+    directly reviewed this PR and its [predecessor][Bitcoin Core
+    #17977], and many others participated in developing and reviewing
+    the underlying research, the BIPs, the related code in libsecp256k1,
+    and other parts of the system.  There is still more work to do, but
+    getting the code properly reviewed was arguably the most critical
+    step---so we extend our most grateful appreciation to everyone who
+    helped achieve this milestone.
+
+- [Bitcoin Core #19988][] refactors the logic Bitcoin Core uses to
+  request transactions from its peers.  The major changes described by
+  the PR description include: Bitcoin Core will now more strongly prefer
+  requesting transactions from *outbound peers*---peers where the
+  connection was opened by the local node; Bitcoin Core will now request
+  an unlimited number of transactions from a peer rather than stopping
+  at 100 (though slow responders won't be preferred); and Bitcoin Core
+  won't keep track of as many pending transaction requests as before,
+  the previous limit being seen as excessive.  The change should also
+  make it easier to test that the transaction request logic is working
+  as expected, with multiple tests for that included in the merged PR.
```

Comment:
'multiple tests' is selling this PR short:

- an easy-to-review naive implementation and fuzz tests demonstrating the equivalence between the naive and efficient implementations
- almost 100% unit test coverage with
- multiple functional tests for testing the high-level functionality.

This is probably now the best tested code in Bitcoin Core!